### PR TITLE
Use a cache instead of sync.Pool in script processor

### PR DESCRIPTION
**Pull request by adriansr**
## What does this PR do?

Updates the script processor to keep a simple cache of Javascript VM sessions instead of relying on sync.Pool for this task.

## Why is it important?

The reasoning behind this change is to avoid sync.Pool discarding and re-allocating new sessions every garbage collection cycle. For large Javascript pipelines, allocating a new session is a very expensive operation that can take hundreds of milliseconds, even seconds to complete. This has a severe impact in ingestion rates.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~~I have made corresponding changes to the documentation~~
- [ ] ~~I have made corresponding change to the default configuration files~~
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~
- [ ] ~~I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

### CODE REVIEW

**Comment by adriansr** [comment]
@andrewkroh I was tempted to add another flag to control if it should be blocking if there are no instances available, or just the current approach of allocating a spare one.

I'm not sure what the implications will be in the Winlogbeat use case.

**Comment by adriansr** [comment]
Jenkins, run tests please

File: libbeat/processors/script/docs/script.asciidoc [hunk 1]
```diff
@@ -113,6 +113,9 @@ is interrupted. You can set this option to prevent a script from running for
 too long (like preventing an infinite `while` loop). By default there is no
 timeout.
 
+`max_cached_sessions`:: This sets the maximum number of Javascript VM sessions
```
Comment (andrewkroh) [inline libbeat/processors/script/docs/script.asciidoc:116]: This should document the default.

**Comment by adriansr** [comment]
CI errors unrelated

**Pull request merged**